pentaho-requirejs-osgi-manager/core/impl/src/main/java/org/pentaho/requirejs/impl/listeners/RequireJsBundleListener.java

```diff
@@ -102,22 +102,24 @@ public void destroy() {
 
   @Override
   public void bundleChanged( BundleEvent bundleEvent ) {
-    final Bundle bundle = bundleEvent.getBundle();
+    if ( this.isListenerActive() ) {
+      final Bundle bundle = bundleEvent.getBundle();
 
-    final int bundleEventType = bundleEvent.getType();
+      final int bundleEventType = bundleEvent.getType();
 
-    boolean shouldInvalidate = false;
+      boolean shouldInvalidate = false;
 
-    if ( bundleEventType == BundleEvent.STARTED ) {
-      shouldInvalidate = addBundle( bundle );
-    } else if ( bundleEventType == BundleEvent.UNINSTALLED
-        || bundleEventType == BundleEvent.UNRESOLVED
-        || bundleEventType == BundleEvent.STOPPED ) {
-      shouldInvalidate = removeBundle( bundle );
-    }
+      if ( bundleEventType == BundleEvent.STARTED ) {
+        shouldInvalidate = addBundle( bundle );
+      } else if ( bundleEventType == BundleEvent.UNINSTALLED
+          || bundleEventType == BundleEvent.UNRESOLVED
+          || bundleEventType == BundleEvent.STOPPED ) {
+        shouldInvalidate = removeBundle( bundle );
+      }
 
-    if ( shouldInvalidate ) {
-      this.requireJsConfigManager.invalidateCachedConfigurations();
+      if ( shouldInvalidate && this.isListenerActive() ) {
```

Comment:
Need to test `isListenerActive` again? Are we trying to guard against multiple threads?

Reply:
Exactly. The fix consists in wrapping the possible NPE in the try-catch block (not preventing it, as `this.serviceRegistrationMap` can become null at any moment) and also don't invalidate the cache if the listener is not active at that moment.

In reality the first `isListenerActive` test is the one not strictly necessary, it's just for short-circuiting earlier.